functions/go/set-setter/go.mod

```diff
@@ -0,0 +1,10 @@
+module github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/set-annotation
```

Comment:
This says `set-annotation`

Also, not fan of  `set-setter` as the name, it sounds clumsy.

We should also rethink the convention for other functions.

One option is to have them be verbs:

set-label -> label
set-annotations -> annotate
set-setter -> ?

Some things to think about, would we ever have separate functions for other CRUD oeprations, e.g. `delete-label`?

@Shell32-Natsu

Reply:
I don't think we have ever had function like delete label. But IMO adding a `set-` prefix makes it less confusing.

Reply:
K8s apiserver supports 3 different formats:
- k8s' own Strategic Merge Patch (SMP)
- JSON merge patch
- JSON patch

In SMP and JSON merge patch, it is allowed to set a field as `null` to clear it. e.g.
```yaml
metadata:
  annotations:
    foo: null # this field will be removed after applying the patch
```

I have tried to use something similar with set-annotation fn:
```
annotation_name: fruit
annotation_value: null
```

It complains: `failed to add annotation [configmanagement.gke.io/namespace-selector: ]: annotation_name and annotation_value cannot be emptyerror: yaml: line 75: mapping values are not allowed in this context`

I wonder if it makes sense to support it here

Reply:
to avoid the stutter, maybe `apply-setter`?

Reply:
I think this needs a broader discussion on listing all the possible functions and names. Also the function config naming.

Reply:
We may need a broader discussion on function naming.
But `set-annotation` is definitely wrong here. Let's use `apply-setter` for now.

Reply:
Yes. `set-annotation` is no longer present in the PR.